### ISSUE TITLE
Suggest `#[derive(PartialEq)]` on E0369

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#85845]
When `PartialEq` isn't implemented for a type, like this

```
error[E0369]: binary operation `==` cannot be applied to type `&Group`
  --> src/tests/a.rs:19:5
   |
19 |     assert_eq!(result.unwrap(), &group);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     &Group
   |     &Group
   |
   = note: an implementation of `std::cmp::PartialEq` might be missing for `&Group`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

We should suggest `#[derive(PartialEq)]` on the appropriate type if possible.

_Taken from https://blog.frankel.ch/start-rust/1/_

**Comment by ccgauche** [comment]
I'm going to try to fix that as my first contribution!

**Comment by jackh726** [comment]
Fixed in #86943
